### PR TITLE
Pro Dashboard: Add a reusable contact list component to unify the implementation of email and SMS in notification settings.

### PR DESCRIPTION
**Pull request by jkguidaven**
This is part of my ongoing refactoring of the Email and Phone number editor to have shared codes.

This PR adds a reusable Contact List component to unify the implementation of the Email and SMS contact list in the Notification settings modal.

Screen Shot 2023-06-23 at 4 16 55 PM

Related to 1204774821045518-as-1204802449809227

## Testing Instructions

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

* Run `git checkout add/downtime-monitoring-contact-list-common-component` and `yarn start-jetpack-cloud`
* Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard.
* Enable monitor if not enabled already.
* Click on the clock icon next to the monitor toggle.
* Confirm that the email and SMS contact list work as expected.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/index.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		onAction?.();
+	};
+
+	return (
+		<>
+			{ items.length === 0 && (
+				<div className="margin-top-16">
+					<AlertBanner type="warning">
+						{ type === 'email'
+							? translate( 'You need at least one email address' )
```
Comment (yashwin) [inline client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/index.tsx]: We won't need this for `email`.

File: client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+						item={ item }
+						onAction={ onAction }
+						recordEvent={ recordEvent }
+						showVerifiedBadge={ getContactItemValue( type, item ) === verifiedItemKey }
+					/>
+				) ) }
+
+				<Button compact className="contact-list__button" onClick={ onAddContact }>
+					<Icon size={ 18 } icon={ plus } />
+					{ type === 'email' ? translate( 'Add email address' ) : translate( 'Add phone number' ) }
```
Comment (yashwin) [inline client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/index.tsx]: Can we also check if the `type` is `phone` here? Let's not have a default one considering we will have other modes of notifications.

File: client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/utils.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		return ( item as StateMonitorSettingsEmail ).email;
+	}
+	return ( item as StateMonitorSettingsSMS ).phoneNumberFull;
+};
+
+export const getContactActionEventName = (
+	type: 'email' | 'phone',
+	action: AllowedMonitorContactActions
+) => {
+	const EVENT_NAMES =
```
Comment (yashwin) [inline client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/utils.ts]: Can we also check if the `type` is `phone` here? Let's not have a default one considering we will have other modes of notifications. 

Something like

```
EVENT_NAMES = {
  email: {
   }
   phone: {
   }
}
```

File: client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/index.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+					<AlertBanner type="warning">
+						{ type === 'email'
+							? translate( 'You need at least one email address' )
+							: translate( 'You need at least one phone number' ) }
+					</AlertBanner>
+				</div>
+			) }
+
+			<div className="contact-list">
+				{ items.map( ( item: StateMonitorSettingsEmail | StateMonitorSettingsSMS ) => (
```
Comment (yashwin) [inline client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/index.tsx]: I don't think we have to specify the type here.

File: client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/utils.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	AllowedMonitorContactActions,
+	StateMonitorSettingsEmail,
+	StateMonitorSettingsSMS,
+} from '../../sites-overview/types';
+
+export const getContactItemValue = (
+	type: 'email' | 'phone',
+	item: StateMonitorSettingsEmail | StateMonitorSettingsSMS
+) => {
+	if ( type === 'email' ) {
```
Comment (yashwin) [inline client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-list/utils.ts:12]: Can we also check if the `type` is phone here? Let's not have a default one considering we will have other modes of notifications.

**Review by yashwin — COMMENTED** [review]
Thanks for working on creating the common components. 

I found an issue here

I don't see any options when I click the `...` icon. Is that expected? 

Screenshot 2023-06-23 at 3 01 51 PM


I have also left some comments.

**Comment by jkguidaven** [comment]
> Thanks for working on creating the common components.
> 
> I found an issue here
> 
> I don't see any options when I click the `...` icon. Is that expected?
> 
> Screenshot 2023-06-23 at 3 01 51 PM
> I have also left some comments.

@yashwin Thanks for reviewing the PR. 🙏  I addressed all the PR comments and hope I didn't miss anything. 

Thanks for catching the popover menu bug. For some reason, it was working on my local machine until I stopped running the Calypso and doing `yarn start-jetpack-cloud` again.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/types.ts [hunk 6]
```diff
@@ -328,8 +328,10 @@ export type MonitorSettingsContact = Partial< MonitorSettingsEmail > &
 
 export type AllowedMonitorContactActions = 'add' | 'verify' | 'edit' | 'remove';
 
+export type AllowedMonitorContactTypes = 'email' | 'sms';
```
Comment (jkguidaven) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/types.ts:331]: I also added allowable contact types and set them to 'email' and 'sms' to match our endpoint. This makes sure we are using the correct strings.

**Review by yashwin — APPROVED** [review]
LGTM 🚀 Great job on this, @jkguidaven.

**Pull request merged**